src/Features/Core/Portable/EditAndContinue/ActiveStatementsMap.cs

```diff
@@ -2,34 +2,291 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System;
 using System.Collections.Generic;
 using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Text;
 using Microsoft.VisualStudio.Debugger.Contracts.EditAndContinue;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.EditAndContinue
 {
-    internal readonly struct ActiveStatementsMap
+    internal sealed class ActiveStatementsMap
     {
         public static readonly ActiveStatementsMap Empty =
-            new(ImmutableDictionary<DocumentId, ImmutableArray<ActiveStatement>>.Empty, ImmutableDictionary<ManagedInstructionId, ActiveStatement>.Empty);
+            new(ImmutableDictionary<string, ImmutableArray<ActiveStatement>>.Empty,
+                ImmutableDictionary<ManagedInstructionId, ActiveStatement>.Empty);
+
+        public static readonly Comparer<ActiveStatement> Comparer =
+            Comparer<ActiveStatement>.Create((x, y) => x.FileSpan.Start.CompareTo(y.FileSpan.Start));
+
+        public static readonly Comparer<(ManagedActiveStatementDebugInfo, SourceFileSpan, int)> s_infoSpanComparer =
+            Comparer<(ManagedActiveStatementDebugInfo, SourceFileSpan span, int)>.Create((x, y) => x.span.Start.CompareTo(y.span.Start));
```

Comment:
Since this is public, presumably should be `InfoSpanComparer`?

Reply:
Should be private